AX: Add test exposing missing space in RTL text after ENABLE(AX_THREAD_TEXT_APIS)

gr3g-apple-com opened the pull request:
#### da81eb616f32a82523ea8d620d838213503d130b
<pre>
AX: Add test exposing missing space in RTL text after ENABLE(AX_THREAD_TEXT_APIS)
<a href="https://bugs.webkit.org/show_bug.cgi?id=292458">https://bugs.webkit.org/show_bug.cgi?id=292458</a>
<a href="https://rdar.apple.com/150549706">rdar://150549706</a>

Reviewed by Tyler Wilcock.

This change adds a new layout test for bidirectional text using the dir property. The test gets a string between text markers, and verifies the length, and that the string value matches the expected string. Currently the expected string checking fails with ENABLE(AX_THREAD_TEXT_APIS), due to a dropped space character.

* LayoutTests/accessibility-isolated-tree/TestExpectations:
* LayoutTests/accessibility/text-marker/text-marker-bidi-inline-cite-expected.txt: Added.
* LayoutTests/accessibility/text-marker/text-marker-bidi-inline-cite.html: Added.

Canonical link: <a href="https://commits.webkit.org/294531@main">https://commits.webkit.org/294531@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cba6bd9dfa8f0dde35e230eadef9444cfb1970e5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102179 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21846 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12162 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107338 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52815 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22155 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30354 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77757 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34752 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105185 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17139 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92240 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58092 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16968 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10266 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52173 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10339 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109714 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29311 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21608 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86733 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29673 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88443 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86317 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21961 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31126 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8843 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/23553 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29239 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/29050 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32373 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30609 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

minorninth commented on LayoutTests/accessibility/text-marker/text-marker-bidi-inline-cite.html:
```diff
@@ -0,0 +1,35 @@
+var output = "This test verifies that a paragraph with ltr text in English, containining a cite element with dir=\"rtl\", containing a span element with dir=\"ltr\" is read correctly by accessibility.\n\n";
+
+if (window.accessibilityController) {
+    var paragraph = accessibilityController.accessibleElementById("content").childAtIndex(0);
```
I think this is actually the first static text element inside the paragraph, not the paragraph element.

gr3g-apple-com replied:
Thx, good point. I renamed this (and related variables).

minorninth commented on LayoutTests/accessibility/text-marker/text-marker-bidi-inline-cite.html:
```diff
@@ -0,0 +1,35 @@
+    output += expect("paragraph.stringForTextMarkerRange(textMarkerRange).trim()", "'The alphabet is'")
+
+    var citeRTL = accessibilityController.accessibleElementById("content").childAtIndex(1);
+    var textMarkerRange = paragraph.textMarkerRangeForElement(citeRTL);
```
Nit: you already declared textMarkerRange above. I'd recommend using const for everything, and make a new variable name for this one, like citeTextMarkerRange.

gr3g-apple-com replied:
`const`/`let` don't work here for any variable passed from a LayoutTest into an `expect` bc of scope

minorninth reviewed (commented):
lgtm!

gr3g-apple-com committed:
AX: Add test exposing missing space in RTL text after ENABLE(AX_THREAD_TEXT_APIS)
https://bugs.webkit.org/show_bug.cgi?id=292458
rdar://150549706

Reviewed by Tyler Wilcock.

This change adds a new layout test for bidirectional text using the dir property. The test gets a string between text markers, and verifies the length, and that the string value matches the expected string. Currently the expected string checking fails with ENABLE(AX_THREAD_TEXT_APIS), due to a dropped space character.

* LayoutTests/accessibility-isolated-tree/TestExpectations:
* LayoutTests/accessibility/text-marker/text-marker-bidi-inline-cite-expected.txt: Added.
* LayoutTests/accessibility/text-marker/text-marker-bidi-inline-cite.html: Added.

Canonical link: https://commits.webkit.org/294531@main
